src/molotovProviderBase.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+
+import type molotov from './molotov';
+
+import type {
+  molotovConfig,
+  overrideConfig,
+  pluginsList,
+  ProviderBase,
+  supersNameSpace,
+} from './types/molotov';
+
+// User overrides are passed in targeted to a hierarchical namespace.
+const _ = require('lodash');
+const validator = require('./validateConfig');
+
+module.exports = class MolotovProviderBase<T: string> implements ProviderBase<T> {
+  target: T
```

Comment:
Is `T` a flow defined type? If not can it be more descriptive?

Reply:
It is a generic. We could make it a literal maybe?

Reply:
("molotovPlugins" | "supersNameSpace") 👍

Reply:
Oh ok, I just don't know flow that well. I think it's fine 😄